src/js/controllers/step1/LottoController.js

```diff
@@ -0,0 +1,219 @@
+      winningAmount: 5_000,
+      numOfWinTicket: 0,
+    },
+    {
+      winningCount: 4,
+      hasToWinBonus: false,
+      winningAmount: 50_000,
+      numOfWinTicket: 0,
+    },
+    {
+      winningCount: 5,
+      hasToWinBonus: false,
+      winningAmount: 1_500_000,
+      numOfWinTicket: 0,
+    },
+    {
+      winningCount: 5,
+      hasToWinBonus: true,
+      winningAmount: 30_000_000,
+      numOfWinTicket: 0,
+    },
+    {
+      winningCount: 6,
+      hasToWinBonus: false,
+      winningAmount: 2_000_000_000,
+      numOfWinTicket: 0,
+    },
+  ];
+
+  constructor(readline) {
+    this.#readline = readline;
+    this.#consoleInput = new ConsoleInput(readline);
+  }
+
+  validateMoney(money) {
+    if (isNaN(money)) {
+      throw new Error("금액은 숫자 타입이어야 합니다.");
+    }
+
+    if (money < 1000) {
+      throw new Error("금액은 1000원 이상이어야 합니다.");
+    }
+
+    if (money % 1000 > 0) {
+      throw new Error("금액은 1000원 단위여야 합니다.");
+    }
+  }
+
+  validateNumber(number) {
+    if (isNaN(number)) throw new Error("번호는 숫자 타입이어야 합니다.");
+
+    if (number < 1 || number > 45) throw new Error("번호는 1 이상 45 이하 숫자여야 합니다.");
+  }
+
+  validateWinningNumbers(winningNumbers) {
+    winningNumbers.forEach((number) => {
+      this.validateNumber(number);
+    });
+
+    if (winningNumbers.length !== 6) {
+      throw new Error("당첨 번호는 6개의 숫자여야 합니다.");
+    }
+
+    if (new Set(winningNumbers).size !== winningNumbers.length) {
+      throw new Error("당첨 번호는 중복되면 안됩니다.");
+    }
+  }
+
+  validateBonusNumber(winningNumbers, bonusNumber) {
+    this.validateNumber(bonusNumber);
+
+    if (winningNumbers.indexOf(bonusNumber) !== -1)
+      throw new Error("보너스 번호는 당첨 번호와 중복되면 안됩니다.");
+  }
+
+  async startLotto() {
+    const money = await this.readMoney();
+    this.#money = money;
+    const lottoTickets = this.issueLottoTickets(money);
+
+    this.printLottoTickets(lottoTickets);
+
+    const { winningNumbers, bonusNumber } = await this.readNumbers();
+    this.printWinningStatistics({ lottoTickets, winningNumbers, bonusNumber });
+
+    this.#readline.close();
+  }
+
+  async readMoney() {
+    const money = await this.#consoleInput.readline("구입금액을 입력해 주세요.");
+    this.validateMoney(money);
+
+    return Number(money);
+  }
+
+  async readNumbers() {
+    const winningNumbersStr = await this.#consoleInput.readline("당첨 번호를 입력해주세요.");
+
+    const winningNumbers = winningNumbersStr.split(",").map(Number);
+    this.validateWinningNumbers(winningNumbers);
+
+    const bonusNumber = Number(await this.#consoleInput.readline("보너스 번호를 입력해 주세요."));
+    this.validateBonusNumber(winningNumbers, bonusNumber);
+
+    return { winningNumbers, bonusNumber };
+  }
+
+  issueLottoTickets(money) {
+    const lottoTickets = [];
+    const NumberOfLotto = Math.floor(money / LottoTicket.price);
+
+    for (let i = 0; i < NumberOfLotto; i++) {
+      lottoTickets.push(new LottoTicket(this.drawLottoNumbers()));
+    }
+
+    return lottoTickets;
+  }
```

Comment:
아래처럼 작성해보는건 어떨까요?

next-step 과정을 진행하는 동안 의식적으로 `for`문 사용을 멀리하시는 노력을 해보셔도 좋을거 같습니다!

```suggestion
  issueLottoTickets(money) {
    const NumberOfLotto = Math.floor(money / LottoTicket.price);
    
    return Array.from({ length: NumberOfLotto }, () => new LottoTicket(this.drawLottoNumbers()));
  }
```